packages/core/src/elements/FormFieldElement.ts

```diff
@@ -0,0 +1,360 @@
+    super.firstUpdated(changedProperties);
+    this.addEventListener('focus', this.onFocus);
+  }
+
+  /**
+   * Handles the focus event
+   * @param event Focus event
+   * @returns {void}
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected onFocus (event: FocusEvent): void {
+    // Always recalculate inherited arias on focus and these may change from last time
+    this.setInheritedAria();
+  }
+
+  /**
+   * Set inherited aria properties
+   * @returns {void}
+   */
+  protected setInheritedAria (): void {
+    this.inputAriaLabel = inputLabel(this);
+    this.inputAriaDescription = inputDescription(this);
+    // no need for required or error here, as it is sufficient to recalculate these on attributeChanged
+  }
+
+  /**
+   * Notify error if it has changed
+   * @param hasError true if the element has an error
+   * @returns {void}
+   */
+  protected notifyErrorChange (hasError: boolean): void {
+    if (this.error !== hasError) {
+      this.error = hasError;
+      this.notifyPropertyChange('error', this.error);
+    }
+  }
+
+  /**
+   * Runs on input element `input` event
+   * @param event `input` event
+   * @returns {void}
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected onInputInput (event: InputEvent): void {
+    // nothing
+  }
+
+  /**
+   * Runs on input element `change` event
+   * @param event `change` event
+   * @returns {void}
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected onInputChange (event: InputEvent): void {
+    // nothing
+  }
+
+  /**
+   * Decorate `<input>` element with common properties:
+   * aria-hidden="true" - always true. Required for onLoad screen read. this is always true for Chrome. Other browsers may need to tweak
+   * aria-label - calculated from `aria-label`, `aria-labelledby` and `label[for="<element.id>"]`
+   * aria-description - calculated from `aria-description` or `aria-describedby`
+   * aria-invalid="true|false" - calculated on based on `error` state
+   * aria-required="true|false" - calculated on based on `aria-required`
+   * placeholder - placeholder text
+   * readonly - indicates whether the user can interact and still show value with the element
+   * disabled - indicates whether the user can interact with the element
+   * autocomplete="off" - always off as is not supported within shadow root
+   * @input - Listener for `input` event. Runs `this.onInputInput`
+   * @change - Listener for `change` event. Runs `this.onInputChange`
+   * @returns template map
+   */
+  protected get decorateInputMap (): TemplateMap {
+    return {
+      'aria-hidden': 'true',
+      'aria-label': this.inputAriaLabel,
+      'aria-description': this.inputAriaDescription,
+      'aria-invalid': this.error ? 'true' : 'false',
+      'aria-required': this.inputAriaRequired ? 'true' : 'false',
+      'placeholder': this.placeholder || null,
+      'readonly': this.readonly ? '' : null,
```

Comment:
This would be a lot nicer if we supported booleans for boolean attributes. It then follows the inbuilt classMap.